lib/batch/subject.csv.rb

```diff
@@ -0,0 +1,22 @@
+csv = CSV.open('/home/deploy/all_gf_subject.csv', 'wb')
+terms =  GalterGenericFilePresenter.terms + [:depositor, :date_uploaded, :date_modified, :read_access_group]
+
+csv << terms.map(&:to_s).map {|t| I18n.t(:simple_form)[:labels][:generic_file][t] || t.to_s.titleize }
+
+ActiveFedora::SolrService.query('has_model_ssim:GenericFile', { rows: 99999 }).each.with_index do |gf, idx|
+  gfmod = gf.each.map {|k, v| [k.gsub(/_[a-z]+\z/, ''), v] }.to_h
+  csv << terms.map {|term|
+    t = term.to_s
+    (gfmod[t].is_a?(Array) ? gfmod[t].reject(&:blank?).compact.join(' ; ') : gfmod[t]).to_s
+  }
+  puts "Index: #{idx}" if idx % 100 == 0
+end
+ActiveFedora::SolrService.query('has_model_ssim:Page', { rows: 99999 }).each.with_index do |gf, idx|
```

Comment:
Does this limit the total rows or paginates them? If the former, we know for sure that we have less than 100,000 records?

Reply:
It limits the rows. Without checking somehow before every time we run this script, we will not know for sure if we have less than this parameter. There is a plan to come up with a different solution to handle this, so we can address this then.